### PR TITLE
Pinterest v5 user onboarding integration.

### LINKED ISSUE #757: Complete the onboarding for the user.
From the documentation:

```Merchant connect/disconnect
The very first endpoints you will use are to connect the user to the commerce integration on the Pinterest side. You will need to reference the metadata retrieved from the pop UI step above.

[Create commerce integration](https://developers.pinterest.com/docs/api/v5/#operation/integrations_commerce/post) with connecting user

You will also be responsible for [disconnecting/deleting the user from the commerce integration](https://developers.pinterest.com/docs/api/v5/#operation/integrations_commerce/del)

You may also [retrieve and update](https://developers.pinterest.com/docs/api/v5/#tag/integrations) when appropriate.
Domain/Website verification
Below are the endpoints that you can use to complete the website verification steps

Useful to establish the method and get the code
[Get the verification method code](https://developers.pinterest.com/docs/api/v5/#operation/website_verification/get)

Useful to verify the website once method and code is established
[Verify the user’s website](https://developers.pinterest.com/docs/api/v5/#operation/verify_website/update) with method and code

See all the user’s websites
[Get the user’s website](https://developers.pinterest.com/docs/api/v5/#operation/user_websites/get) (verified and unverified)

Delete  a specific verified website
[Delete a verified website](https://developers.pinterest.com/docs/api/v5/#operation/unverify_website/delete)

Note: This can be [done manually by the user](https://help.pinterest.com/en/business/article/claim-your-website) and they may already have a verified website.```

### PR DESCRIPTION
**Pull request by message-dimke**
### Changes proposed in this Pull Request:

Closes #757 .

Integrating Pinterest extension onboarding process with Pinterest API v5. Porting API endpoints from elder versions to v5, changing onboarding process for the plugin.

### Detailed test instructions:

0. Disconnect from Pinterest account or make a fresh installation of Pinterest extension.
1. Checkout `pinterest-v5-integration-user-connect-wizard-integration` branch.
2. Follow the connection procedure.
3. You should successfully connect to Pinterest.

Being connected you could also try and sync Automatic Enhanced Match support. In order to test that, go to your Pinterest dashboard, find the tag and turn the switch on/off. Syncing setting inside the plugin should reflect the changes made at Pinterest side.

Disable the setting:

_2__Pinterest

Sync and check the extension got it synchronised.

Settings_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce

Being connected you could check auto tag creation for ad accounts which have no tag.
- create ad account

_2__Pinterest

- go to plugin connection settings and pick an ad account which has no tag from the dropdown. 

Connection_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce

- observe a tag with a default name got created.

Cursor_and_Connection_‹_Pinterest_‹_Marketing_‹_WooCommerce_‹_WordPress_Pinterest_—_WooCommerce

4. Disconnect the extension from Pinterest.

### Changelog entry

> Upgrade - Pinterest API v5 connection process integration.
> Upgrade - Pinterest API v5 Settings and Connection tabs integration.

### CODE REVIEW

**Comment by budzanowski** [comment]
When the url validation fails the behaviour is suboptimal :) 
image

**Comment by message-dimke** [comment]
Hey, @budzanowski ! Adjusted the UI to have a button to manually control retries in case the initial verification failed for some reason. Now  the retry must not start automatically and wait for user command instead. My bad.

File: assets/source/setup-guide/app/steps/SetupAccount.js
Comment (message-dimke) [inline assets/source/setup-guide/app/steps/SetupAccount.js:1]: All the changes to the file are just lint:js fixes.

**Comment by message-dimke** [comment]
Closing this one if favour of #778

**Pull request closed without merging**